File: examples/rutaflow/ios/LocationPolicy.swift
import Foundation

enum RouteActivity { case stopped, walkingToDoor, driving }

struct LocationPolicy: Equatable {
    let minimumInterval: Duration
    let distanceFilterMeters: Double

    static func forRoute(activity: RouteActivity, batteryLevel: Double) -> Self {
        precondition((0...1).contains(batteryLevel))
        if batteryLevel <= 0.15 {
            return .init(minimumInterval: .seconds(120), distanceFilterMeters: 200)
        }
        return switch activity {
        case .stopped: .init(minimumInterval: .seconds(180), distanceFilterMeters: 250)
        case .walkingToDoor: .init(minimumInterval: .seconds(15), distanceFilterMeters: 10)
        case .driving: .init(minimumInterval: .seconds(30), distanceFilterMeters: 50)
        }
    }
}

File: examples/rutaflow/android/LocationPolicy.kt
package academy.rutaflow.location

data class LocationPolicy(val intervalMillis: Long, val minDistanceMeters: Float)

enum class RouteActivity { STOPPED, WALKING_TO_DOOR, DRIVING }

fun locationPolicy(activity: RouteActivity, batteryPercent: Int): LocationPolicy {
    require(batteryPercent in 0..100)
    if (batteryPercent <= 15) return LocationPolicy(intervalMillis = 120_000, minDistanceMeters = 200f)
    return when (activity) {
        RouteActivity.STOPPED -> LocationPolicy(180_000, 250f)
        RouteActivity.WALKING_TO_DOOR -> LocationPolicy(15_000, 10f)
        RouteActivity.DRIVING -> LocationPolicy(30_000, 50f)
    }
}
